src/target/riscv_debug.c

```diff
@@ -304,6 +304,30 @@ void riscv_dmi_init(riscv_dmi_s *const dmi)
 	do {
 		/* Read out the DM's status register */
 		uint32_t dm_status = 0;
+		uint32_t dm_control = 0;
+		bool dm_active = false;
+		static uint8_t counter = 0;
```

Comment:
Why is this `static`? This reads like it should be reset each entry to this function. Otherwise, after 100 scans this function permanently fails.